### PR TITLE
Issue 1 create an entry

### PR DESCRIPTION
**Pull request by DavidKyong**
Recording 2023-05-30 at 14 59 03

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "uname -a",
+
+	// Configure tool-specific properties.
+	// "customizations": {},
+
+	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+	// "remoteUser": "root"
```
Comment (shawnkost) [inline .devcontainer/devcontainer.json]: Nothing should have changed inside of this `devcontainer.json` file. Please revert back to what was there before.

File: css/layout.css [hunk 2]
```diff
@@ -6,6 +6,12 @@
   max-width: 575px;
   margin: 0 auto;
   padding: 0 1.5rem;
+  background-color: rgb(240, 240, 240);
+}
+
+form {
+  font-family: "open sans";
```
Comment (shawnkost) [inline css/layout.css]: Need to set a backup font

File: css/styles.css [hunk 3]
```diff
@@ -0,0 +1,58 @@
+.button {
+  justify-content: right;
+  display: flex;
+  margin-top: 1rem;
+  margin-bottom: 1rem;
+}
+
+header {
+  font-family: "proza libre";
```
Comment (shawnkost) [inline css/styles.css]: Need to set a backup font

File: css/styles.css [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  padding-left: 1.5rem;
+  padding-bottom: 0.25rem;
+}
+
+@media screen and (min-width: 768px) {
+  .tabTitle {
+    margin-right: -2.25rem;
+    margin-left: -2.25rem;
+    padding-left: 2.25rem;
+  }
```
Comment (shawnkost) [inline css/styles.css]: Be careful when using negative margin. If you're using negative margin, then you're probably doing something "hacky" to get this to work

File: index.html [hunk 5]
```diff
@@ -14,17 +14,73 @@
...
+        </div>
+      </div>
+
+      <div class="row">
+        <div class="column-half">
+          <div class="picTitle">
+            <h2>New Entry</h2>
+          </div>
+        </div>
+      </div>
```
Comment (shawnkost) [inline index.html]: This should be inside your `div` with `data-view="entry-form"`

**Comment by DavidKyong** [comment]
Recording 2023-05-30 at 15 15 01

File: index.html [hunk 6]
```diff
@@ -14,17 +14,73 @@
...
+        <div class="row">
+          <div class="column-half">
+            <div class="picture">
+              <img src="/images/placeholder-image-square.jpg" alt="placeholder icon">
+            </div>
+          </div>
+          <div class="column-half">
+            <div>
+              <label for="title">Title</label><br>
+              <input id="title" name="pageTitle" required>
```
Comment (shawnkost) [inline index.html]: Needs a `type` attribute

File: index.html [hunk 7]
```diff
@@ -14,17 +14,73 @@
...
+            </div>
+            <div>
+              <label for="url">Photo URL</label><br>
+              <input id="url" name="photoURL" required>
```
Comment (shawnkost) [inline index.html]: Needs a `type` attribute

File: index.html [hunk 8]
```diff
@@ -14,17 +14,73 @@
...
+            <div>
+              <label for="notes">Notes</label><br>
+
+              <textarea id="notes" cols="40" rows="10" required name="message"></textarea>
+            </div>
+          </div>
+          <div class="column-full">
+            <div class="button">
+
+              <label for="save"></label><br>
```
Comment (shawnkost) [inline index.html]: Don't need a label here, since there's no text inside

File: js/.devcontainer/devcontainer.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+      "extensions": [
+        "thebearingedge.lfz-code"
+      ]
+    }
+  },
+  "postCreateCommand": "[ ! -f package-lock.json ] || npm ci; sudo chown -R dev:dev ~/.ssh && chmod 700 ~/.ssh && chmod 600 ~/.ssh/* && chmod 644 ~/.ssh/*.pub",
+  "otherPortsAttributes": {
+    "onAutoForward": "silent"
+  }
+}
```
Comment (shawnkost) [inline js/.devcontainer/devcontainer.json]: Delete this whole `.devcontainer` folder that's inside of your `js` folder

File: js/main.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+const $inputURL = document.querySelector('#url');
+const $image = document.querySelector('img');
+const $form = document.querySelector('form');
+
+$inputURL.addEventListener('input', function (event) {
+  event.preventDefault();
```
Comment (shawnkost) [inline js/main.js]: No need for `event.preventDefault` here

File: js/main.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  const formValue = {
+    title: $title,
+    url: $url,
+    notes: $message
+  };
+
+  formValue.entryId = data.nextEntryId;
+  data.nextEntryId++;
+
+  data.entries.push(formValue);
```
Comment (shawnkost) [inline js/main.js]: `formValue` should be getting added to the beginning of the array

File: js/main.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+  $image.setAttribute('src', '/images/placeholder-image-square.jpg');
+  event.target.reset();
```
Comment (shawnkost) [inline js/main.js]: `event.target` will refer to the element that was interacted with, which in this case, I believe would be the submit button. I would just do `$form.reset()`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Looking good so far David! A few changes requested here below. Make sure to also fix the background colors I told you about in our messages. Let me know if you have any questions!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**